### PR TITLE
control: refactor nemesis logs

### PR DESCRIPTION
**Pull request by mahjonp**
Signed-off-by: mahjonp <junpeng.man@gmail.com>

### What is changed and how does it work?

Add more logs on nemesis and will retry when recovery from nemesis fails

### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/control/control.go [hunk 1]
```diff
@@ -245,23 +244,32 @@ func (c *Controller) RunSelfScheduled() {
...
+		})
+	}
+	if err := g.Wait(); err != nil {
+		log.Fatalf("run client error, %+v", errors.ErrorStack(err))
 	}
 
-	ncancel()
+	// cancel nCtx and wait nemesis ended
+	nCancel()
+	nemesisWg.Wait()
```
Comment (you06) [inline pkg/control/control.go:271]: Seems this wait group is equal to a channel

File: pkg/control/control.go [hunk 2]
```diff
@@ -378,61 +383,68 @@ LOOP:
...
 			default:
 			}
-
-			log.Printf("begin to run %s nemesis generator", g.Name())
-			ops := g.Generate(c.cfg.Nodes)
-
-			wg.Add(len(ops))
+			var (
+				ops = g.Generate(c.cfg.Nodes)
+				g   errgroup.Group
```
Comment (you06) [inline pkg/control/control.go:388]: It's better to use another variable name instead of overwriting `g`

File: pkg/control/control.go (hunk 1 above)
Comment (mahjonp) [reply pkg/control/control.go:271]: I found that we lost this wait statement in preivous.

File: pkg/control/control.go (hunk 2 above)
Comment (mahjonp) [reply pkg/control/control.go:388]: Good catch, I didn't notice that.

Comment (mahjonp) [reply pkg/control/control.go:388]: Updated.

File: pkg/control/control.go (hunk 1 above)
Comment (you06) [reply pkg/control/control.go:271]: I just feel a little odd about wait group has only one task (it's equal to waiting for a channel), but it's ok to me.

**Pull request merged**